pycode/memilio-epidata/memilio/epidata/getCommuterMobility.py

```diff
@@ -193,199 +221,87 @@ def get_commuter_data(read_data=dd.defaultDict['read_data'],
-                              abs(float(commuter_migration_file.iloc[i][2]) - float(
-                                  commuter_migration_file.iloc[i - 1][2])) == 2)):
-
-                        # auxiliary key of Bundesland (key translated to int starting at zero)
-                        dummy_key = int(commuter_migration_file.iloc[i][2]) - 1
-
-                        # sum population of all counties not explicitly migrated
-                        # from the current gov region migrated from
-                        dummy_pop_sum = 0
-                        for j in range(0, len(state_gov_table[dummy_key])):
-                            # over all government regions not explicitly stated
-                            gov_index = govkey2numlist[state_gov_table[dummy_key][j]]
-                            for k in range(0,
-                                           len(gov_county_table[gov_index])):
-                                # over all counties of the considered gov region
-                                if counties_migratedfrom[gov_index][k] < 1:
+    with progress_indicator.Spinner() as p:
+        for i in range(len(commuter_migration_files)):
+            p.set_message('Progress '+str(i+1)+'/16')
+            commuter_migration_file = commuter_migration_files[i]
+
+            counties_done = []  # counties considered as 'migration from'
+            # current_row = -1  # row of matrix that belongs to county migrated from
+            current_col = -1  # column of matrix that belongs to county migrated to
+            checksum = 0  # sum of county migration from, to be checked against sum in document
+
+            for i in range(0, commuter_migration_file.shape[0]):
+                if (len(str(commuter_migration_file.iloc[i][0])) == 5
+                        and (commuter_migration_file.iloc[i][0]).isdigit()):
+                    checksum = 0
+                    # make zero'd list of counties explicitly migrated to from county considered
+                    # 'implicit' migration means 'migration to' which is summed in a larger regional entity and not given in
+                    # detail per county
+                    counties_migratedfrom = []
+                    for j in range(0, len(gov_county_table)):
+                        counties_migratedfrom.append(
+                            np.zeros(len(gov_county_table[j])))
+
+                    counties_done.append(commuter_migration_file.iloc[i][0])
+                    current_col = countykey2numlist[commuter_migration_file.iloc[i][0]]
+                    curr_county_migratedto = commuter_migration_file.iloc[i][1]
+                    current_key = commuter_migration_file.iloc[i][0]
+                    # migration to itself excluded!
+                    counties_migratedfrom[countykey2govkey[current_key]][
+                        countykey2localnumlist[current_key]] = 1
+
+                if not isinstance(commuter_migration_file.iloc[i][2], float):
```

Comment:
personally, i like commuter_migration_file.iloc[i,2] more than commuter_migration_file.iloc[i][2] :)

Reply:
Me too :) changed it.